src/streamlink/plugins/rumble.py

```diff
@@ -0,0 +1,148 @@
+
+    _SCHEMA_STREAM = {
+        str: validate.all(
+            {
+                "url": validate.url(),
+                "meta": {"bitrate": int},
+            },
+            validate.union_get("url", ("meta", "bitrate")),
+        ),
+    }
+
+    @classmethod
+    def stream_weight(cls, key):
+        weight = cls.QUALITY_WEIGHTS.get(key)
+        if weight:
+            return weight, key
+
+        return super().stream_weight(key)
+
+    def _get_streams(self):
+        if self.matches["channel"]:
+            channel = self.match["channel"]
+            live_uri = self.session.http.get(
+                self._URL_WEB_LIVE.format(channel=channel),
+                schema=validate.Schema(
+                    validate.parse_html(),
+                    validate.xml_xpath_string(
+                        "/html/body/main/section/ol/div[1]/div[2][@class='videostream__footer "
+                        + "videostream__footer--live']/a/@href",
+                    ),
+                ),
+                timeout=5,
+            )
+            if not live_uri:
+                return
+            room_url = f"https://rumble.com{live_uri}"
+        elif self.matches["live"]:
+            room_url = self.url
+        else:
+            return
+        embed_url = self.session.http.get(
+            room_url,
+            schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(
+                    '//script[@type="application/ld+json"]',
+                ),
+                validate.parse_json(),
+                list,
+                validate.length(1),
+                validate.get(0),
+                {
+                    "embedUrl": validate.url(),
+                },
+                validate.get("embedUrl"),
+            ),
+            timeout=5,
+        )
+        match = re.search(r"/embed/([^/]+)/$", embed_url)
+        if not match:
+            return
+        self.id = match.group(1)
+        live_detail = self.session.http.get(
+            self._URL_API_LIVE_DETAIL,
+            params={
+                "v": self.id,
+                "request": self._PARAM_REQUEST,
+                "ver": self._PARAM_VER,
+            },
+            schema=validate.Schema(
+                validate.parse_json(),
+                {
+                    "ua": {
+                        validate.optional("hls"): validate.any(
+                            {
+                                "auto": {
+                                    "url": validate.url(),
+                                },
+                            },
+                            [],
+                        ),
+                        validate.optional("mp4"): validate.any(self._SCHEMA_STREAM, []),
+                        validate.optional("webm"): validate.any(self._SCHEMA_STREAM, []),
+                    },
+                    "title": str,
+                    "author": {"name": str},
+                },
```

Comment:
The HLS multivariant playlist URLs seem to be static:
`https://rumble.com/live-hls/{self.id}/playlist.m3u8`

Is there a need for mp4/webm containers via `HTTPStream`?

Reply:
If the video is in live broadcast mode, then HLS is available. Once the live broadcast ends, HLS is no longer accessible, but MP4 can still be obtained, for example on this page
https://rumble.com/v5680wc-the-rob-carson-show-07092024-hour-1-newsmax-podcasts.html 
However, I have not yet discovered any instances of the use of WebM

Reply:
> The HLS multivariant playlist URLs seem to be static: `https://rumble.com/live-hls/{self.id}/playlist.m3u8`
> 
> Is there a need for mp4/webm containers via `HTTPStream`?

You are right, currently HLS follows this rule, as long as you have obtained self.id, you can already get the HLS URL. I didn't notice this when I was developing.